src/compiler/xslt/invoke-compiled/group-invocation.xsl

```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet xmlns="http://www.w3.org/1999/XSL/TransformAlias"
+                xmlns:x="http://www.jenitennison.com/xslt/xspec"
+                xmlns:xs="http://www.w3.org/2001/XMLSchema"
+                xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                exclude-result-prefixes="#all"
+                version="3.0">
+
+   <!--
+      mode="x:group-invocation"
+      This mode groups the invocation instructions of compiled x:scenario into a single xsl:for-each.
+   -->
+
+   <xsl:template match="document-node()" as="element()*" mode="x:group-invocation">
+      <!-- xsl:call-template elements originating from x:scenario -->
+      <xsl:variable name="scenario-invokers" as="element(xsl:call-template)*"
+         select="xsl:call-template[processing-instruction(origin) eq 'scenario']" />
+
+      <!-- All the invocation instructions originating from x:scenario must be adjacent so that they
+         can be grouped in a single group. -->
+      <xsl:for-each select="
+            node()
+            [not(. intersect $scenario-invokers)]
+            [. >> $scenario-invokers[1]]
+            [$scenario-invokers[last()] >> .]">
+         <xsl:message terminate="yes">
+            <xsl:call-template name="x:prefix-diag-message">
+               <xsl:with-param name="message"
+                  select="'Unexpected node between invocation instructions of compiled scenario.'" />
+            </xsl:call-template>
+         </xsl:message>
+      </xsl:for-each>
+
+      <!-- Group the adjacent xsl:call-template elements originating from x:scenario. The other
+         nodes (actually, xsl:call-template elements (from x:expect) and x:variable elements (from
```

Comment:
Do you mean `xsl:variable` on this line, instead of `x:variable`?

Reply:
Fixed in 4560adf29d88e8aa5bb4605b812ea005bb4688b8.